indexer.ts

```diff
@@ -0,0 +1,743 @@
+      receiver_id: receiverId,
+      timestamp: created_at,
+      type: "Challenge_Payout",
+      action_result: receiverId,
+      tx_hash: receiptId,
+    };
+
+    await context.db.Activity.insert(activity);
+  }
+
+  async function handleListAdminRemoval(args, receiverId, signerId, receiptId) {
+    let data = JSON.parse(args);
+    console.log("removing admin...: ", { ...data }, receiverId);
+    let ts = new Date(
+      Number(BigInt(block.header().timestampNanosec) / BigInt(1000000)) // convert to ms then date
+    )
+    let list = await context.db.List.select({id: receiverId})
+
+    for (const acct in data.admins) {
+      await context.db.ListAdmin.delete({ list_id: list[0].id, admin_id: acct})
+    }
+
+    let activity = {
+      signer_id: signerId,
+      receiver_id: receiverId,
+      timestamp: ts,
+      type: "Remove_List_Admin",
+      tx_hash: receiptId,
+    };
+
+    await context.db.Activity.insert(activity);
+  }
+
+  const GECKO_URL = "https://api.coingecko.com/api/v3";
+  function formatDate(date) {
+    const year = date.getFullYear();
+    const month = String(date.getMonth() + 1).padStart(2, "0");
+    const day = String(date.getDate()).padStart(2, "0");
+
+    return `${day}-${month}-${year}`;
+  }
+
+  // helper function to format
+  function formatToNear(yoctoAmount) {
+    const nearAmount = yoctoAmount / 10 ** 24;
+    return nearAmount;
+  }
+
+  async function handleNewDonations(
+    args,
+    receiverId,
+    signerId,
+    actionName,
+    receiptId
+  ) {
+    let matching_pool = JSON.parse(args).matching_pool
+    if ((actionName == "donate" && receiverId != "donate.potlock.near") || matching_pool) {
+      console.log("calling donate on projects...");
+      return;
+    }
+    console.log("action and recv", actionName, receiverId);
+    let receipt = block
+      .receipts()
+      .filter((receipt) => receipt.receiptId == receiptId)[0];
+    let donation_data = JSON.parse(atob(receipt.status["SuccessValue"]));
+    console.log("result arg...:", donation_data);
+    let donated_at = new Date(donation_data.donated_at || donation_data.donated_at_ms);
+    await context.db.Account.upsert({ id: donation_data.donor_id }, ["id"], []);
+    let endpoint = `${GECKO_URL}/coins/${
+      donation_data.ft_id || "near"
+    }/history?date=${formatDate(donated_at)}&localization=false`;
+    let response = await fetch(endpoint);
+    let data = await response.json();
+    let unit_price = data.market_data?.current_price.usd;
+    if (!unit_price) {
+      console.log("api rate limi?::", data)
+      unit_price = parseFloat(localStorage.getItem("last_price"))
+    }
+    localStorage.setItem("last_price", unit_price.toString()) // store last price incase subsequent price fetch fails,  does it support local storage?
+
+    let total_amount = donation_data.total_amount;
+    let net_amount = (
+      BigInt(donation_data.total_amount) - BigInt(donation_data.protocol_fee)
+    ).toString();
```

Comment:
Pot contract donations contain `net_amount` on `donation_data`

In the case of Donation contract donations, referrer_fee should also be subtracted (if present)